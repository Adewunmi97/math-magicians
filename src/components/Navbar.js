/* eslint-disable no-unused-vars */
import React, { useState } from 'react';
import { Link } from 'react-router-dom';
import './Navbar.css';

function Navbar() {
  const [click, setClick] = useState(false);

  const closeMobileMenu = () => setClick(false);

  return (
    <div>
      <nav className="navbar">
        <div className="navbar-container">
          <Link to="/" className="navbar-logo" onClick={closeMobileMenu}>
            Math magicians
            {' '}
            <i className=" fab fa-typo3" />
          </Link>
          <ul className={click ? 'nav-menu active' : 'nav-menu'}>
            <li className="nav-item">
              <Link to="/" className="nav-links" onClick={closeMobileMenu}>
                Home
              </Link>
            </li>
            <li className="nav-item">
              <Link
                to="/calc"
                className="nav-links"
                onClick={closeMobileMenu}
              >
                Calculator
              </Link>
            </li>
            <li className="nav-item">
              <Link to="/quote" className="nav-links" onClick={closeMobileMenu}>
                Quote
              </Link>
            </li>
          </ul>
        </div>
      </nav>
    </div>
  );
}

export default Navbar;
