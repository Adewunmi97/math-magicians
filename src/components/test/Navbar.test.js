import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { BrowserRouter } from 'react-router-dom';

import Navbar from '../Navbar';

describe('test Navbar functionality', () => {
  it('should have 4 links on page', () => {
    const { container } = render(<Navbar />, { wrapper: BrowserRouter });
    expect(screen.getAllByRole('link')).toHaveLength(4);
    expect(container).toMatchSnapshot();
  });
});
