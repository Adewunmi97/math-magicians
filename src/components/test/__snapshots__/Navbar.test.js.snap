// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Navbar functionality should have 4 links on page 1`] = `
<div>
  <div>
    <nav
      class="navbar"
    >
      <div
        class="navbar-container"
      >
        <a
          class="navbar-logo"
          href="/"
        >
          Math magicians
           
          <i
            class=" fab fa-typo3"
          />
        </a>
        <ul
          class="nav-menu"
        >
          <li
            class="nav-item"
          >
            <a
              class="nav-links"
              href="/"
            >
              Home
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class="nav-links"
              href="/calc"
            >
              Calculator
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class="nav-links"
              href="/quote"
            >
              Quote
            </a>
          </li>
        </ul>
      </div>
    </nav>
  </div>
</div>
`;
